frontend/src/components/Searchbar/useSearchbar.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Hook exposing convenient variables for use with `Searchbar` component.
+ */
+
+import {
+  RefObject,
+  useCallback,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react'
+
+type UseSearchbarReturn = {
+  inputRef: RefObject<HTMLInputElement>
+  isExpanded: boolean
+  handleExpansion: () => void
+  handleCollapse: () => void
+}
+
+export const useSearchbar = ({
+  isInitiallyExpanded = false,
+  isFocusOnExpand = false,
+}: {
+  /**
+   * If `true`, the searchbar will be expanded on initial render.
+   */
+  isInitiallyExpanded?: boolean
+  /**
+   * If `true`, the searchbar will be focused whenever the searchbar is expanded.
+   */
+  isFocusOnExpand?: boolean
+}): UseSearchbarReturn => {
+  const [isExpanded, onExpansionChange] = useState(isInitiallyExpanded)
```

Comment:
isn't the convention to use `is...` and `setIs....`?

Reply:
good catch! probably accidentally renamed